src/app/shared/operators/with-load-more.ts

```diff
@@ -0,0 +1,95 @@
+import { combineLatest, filter, map, Observable, shareReplay, startWith, switchMap } from 'rxjs';
+import { canLoadMoreItems } from '../helpers/load-more';
+import { fetchingState, FetchState, readyState } from '../helpers/state';
+import { mapToFetchState, readyData } from './state';
+
+type FromId = string;
+
+interface WithLoadMore<Data> {
+  state$: Observable<FetchState<Data>>,
+  loadMore$: Observable<{ fromId: FromId } | null>,
+  loadMoreError$: Observable<FetchState<Data>>,
+}
+
+interface WithLoadMoreOptions<Trigger extends { fromId?: FromId | undefined }, Data extends any, T extends any> {
+  trigger$: Observable<Trigger>,
+  fetcher: (trigger: Trigger) => Observable<Data>,
+  limit: number,
+  mapIdFromListItem: (item: T) => FromId,
+  mapListFromData: (data: Data) => T[],
+  mapListToData: (data: Data, list: T[]) => Data,
```

Comment:
Aren't `mapListFromData` and `mapListToData` useful only because the service returning progress matrix returns a "weird" format with columns and rows separated, and so each data "element" not corresponding to one row unlike all other services?
If so, a smarter solution would be to just "fix" the output of the service, i.e., in the service just reorganize the output so that one element is one row. (in the backend output, it could not be done as it would have created many duplicated data, but here, that may be not that important). 
At least, all paging should not be polluted by a service having an uncommon format.